### ISSUE TITLE
groove_playlist_insert misses linking next->prev

### ISSUE THREAD

**Issue by kalhartt** [andrewrk/libgroove#123]
I'm having an issue replacing the first element in a playlist doing something like the following:

````
GroovePlaylist *playlist = ...; // Some playlist with a few items in it
GrooveFile *newItem = ...; // Some new file
groove_playlist_insert(playlist, newItem, 1.0, 1.0, playlist.head);
groove_playlist_remove(playlist, playlist.head);
````

The `next->prev = item` should be outside of the `if( next->prev )` in [playlist.c:L1317](https://github.com/andrewrk/libgroove/blob/master/src/playlist.c#L1317-L1323) right?

**Comment by andrewrk** [comment]
Nice find. I think you're right. I'll fix this in the master branch which will be 5.0.0 - which is not ready for release yet. Would you also like me to fix this in another branch and release 4.3.1?

**Comment by kalhartt** [comment]
That's great, thanks! It's not critical for me, so you don't have to do a release for it.
